### DISCUSSION TITLE
I2E cloned packets

### DISCUSSION THREAD

**Discussion by Arishbiswas** [NIKSS-vSwitch/nikss#113] [Q&A]
hello ,
i am writing a program where i want to do ingress to engress cloning (I2E) and sent the cloned packet only to port 3 and original packet to its  destination, how would i do that, i am getting errors if i use the egress_port field of egress_output_metadata_t , to send the packet out to port 3

the error it shows is:
i2e.p4(125): [--Werror=type-error] error: 'send_to_port(ostd, egress_port)'
        send_to_port(ostd, egress_port);
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ---- Actual error:
  /usr/local/share/p4c/p4include/bmv2/psa.p4(280): Cannot unify 'psa_egress_output_metadata_t' with 'psa_ingress_output_metadata_t'
  struct psa_egress_output_metadata_t {
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  /usr/local/share/p4c/p4include/bmv2/psa.p4(252)
  struct psa_ingress_output_metadata_t {
  
please help me

**Comment by tatry** [comment]
Look at this test case: https://github.com/p4lang/p4c/blob/main/backends/ebpf/tests/p4testdata/clone-i2e.p4

In general you can't change egress port on egress pipeline. It is only doable on ingress. In other words, ingress have to know where to send packet and its copies (in this case configured via `clone session`) in case of i2e clone.

If you really need clone on egress, here you are en example: https://github.com/p4lang/p4c/blob/main/backends/ebpf/tests/p4testdata/etm-clone-e2e.p4.
